full/src/main/java/apoc/mongodb/Mongo.java

```diff
@@ -0,0 +1,118 @@
+package apoc.mongodb;
+
+import apoc.result.LongResult;
+import apoc.result.MapResult;
+import apoc.util.JsonUtil;
+import org.bson.Document;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static apoc.mongodb.MongoDBUtils.Coll;
+import static apoc.mongodb.MongoDBUtils.getDocument;
+import static apoc.mongodb.MongoDBUtils.getMongoColl;
+
+public class Mongo {
+
+    @Context
+    public Log log;
+
+    @Procedure("apoc.mongo.aggregate")
+    @Description("apoc.mongo.aggregate(uri, pipeline, $config) yield value - perform an aggregate operation on mongodb collection")
+    public Stream<MapResult> aggregate(@Name("uri") String uri, @Name("pipeline") List<Map<String, Object>> pipeline, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        MongoDbConfig conf = new MongoDbConfig(config);
+        return executeMongoQuery(uri, conf, 
+                coll -> {
+                    final List<Document> pipelineDocs = pipeline.stream().map(MongoDBUtils::getDocument).collect(Collectors.toList());
+                    return coll.aggregate(pipelineDocs).map(MapResult::new);
+                }, 
+                getExceptionConsumer("apoc.mongo.aggregate", uri, config));
+    }
+
+    @Procedure("apoc.mongo.count")
+    @Description("apoc.mongo.count(uri, query, $config) yield value - perform a count operation on mongodb collection")
+    public Stream<LongResult> count(@Name("uri") String uri, @Name("query") Object query, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        MongoDbConfig conf = new MongoDbConfig(config);
+        return executeMongoQuery(uri, conf, coll -> {
+            long count = coll.count(getDocument(query));
+            return Stream.of(new LongResult(count));
```

Comment:
could this just return a `LongResult`?

Reply:
As a procedure, I have to return a `Stream` because of `org.neo4j.internal.kernel.api.exceptions.ProcedureException: Procedures must return a Stream of records`. 
We could return a `long` by turning it into function, but for consistency with the old procedures, I think it's better this way.

Reply:
Ok. All good. Thanks for explaining!